modules/cart/src/CartManager.php

```diff
@@ -151,8 +154,20 @@ public function removeOrderItem(OrderInterface $cart, OrderItemInterface $order_
     $cart->removeItem($order_item);
     $this->eventDispatcher->dispatch(CartEvents::CART_ORDER_ITEM_REMOVE, new CartOrderItemRemoveEvent($cart, $order_item));
     if ($save_cart) {
+      $this->resetCheckoutFlow($cart);
       $cart->save();
     }
   }
 
+  /**
+   * Resets the checkout flow status.
+   *
+   * @param OrderInterface $cart
+   */
+  private function resetCheckoutFlow(OrderInterface $cart) {
```

Comment:
1) We use protected instead of private.
2) The @param must have a FQCN instead of just OrderInterface, and it needs a description
3) You are reseting the checkoutStep, not the checkoutFlow
4) Your if() will crash if commerce_checkout is not enabled. Instead you need $cart->hasField('checkout_step')